### PR TITLE
Programmer error and logging in PAPI request manager

### PR DESCRIPTION
**Pull request by cjllanwarne**
* Addresses the "programmer error" in JES API manager
* Makes the logging in these classes more usable and traceable
* Replaces strings like `"JES API polling worker"` with strings like `"PAPI request worker"`

### CODE REVIEW

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/api/PipelinesApiRequestManager.scala [hunk 1]
```diff
@@ -177,33 +177,57 @@ class PipelinesApiRequestManager(val qps: Int Refined Positive, requestWorkers:
...
+     Triggered by the StopAndLogSupervisor trait whenever a child (ie supevisee) terminates unexpectedly.
+   */
   override protected def onFailure(terminee: ActorRef, throwable: => Throwable) = {
     // We assume this is a polling actor. Might change in a future update:
     workInProgress.get(terminee) match {
       case Some(work) =>
-        // Most likely due to an unexpected HTTP error, push the work back on the queue and keep going
-        log.error(throwable, s"The JES API worker actor $terminee unexpectedly terminated while conducting ${work.workBatch.tail.size + 1} polls. Making a new one...")
+        // Most likely due to an unexpected HTTP error.
+        // Do some book-keeping, log the error, and then push the work back onto the queue and keep going
```
Comment (mcovarr) [inline supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/api/PipelinesApiRequestManager.scala]: that's the spirit

**Comment by cjllanwarne** [comment]
Sample of a possible new log message thread:
```
[INFO] [...] [.../TestJesApiQueryManager-1262117937] Running with 1 PAPI request workers
[INFO] [...] [.../statusPoller1] received 5 requests from TestJesApiQueryManager-1262117937
[WARN] [...] [.../TestJesApiQueryManager-1262117937] PAPI request worker statusPoller1 terminated. 0 run creation requests, 5 status poll requests, and 0 abort requests will be reconsidered. If any of those succeeded in the cloud before the batch request failed, they might be run twice. Exception details: cromwell.backend.google.pipelines.common.api.PipelinesApiRequestManager$$anonfun$1$$anon$1: PipelinesApiRequestHandler actor termination caught by manager
[INFO] [...] [.../TestJesApiQueryManager-1262117937] PAPI request worker statusPoller1 has been removed and replaced by statusPoller2 in the pool of 1 workers
[INFO] [...] [.../statusPoller2] received 5 requests from TestJesApiQueryManager-1262117937
[WARN] [...] [.../TestJesApiQueryManager-1262117937] PAPI request worker statusPoller2 terminated. 0 run creation requests, 5 status poll requests, and 0 abort requests will be reconsidered. If any of those succeeded in the cloud before the batch request failed, they might be run twice. Exception details: cromwell.backend.google.pipelines.common.api.PipelinesApiRequestManager$$anonfun$1$$anon$1: PipelinesApiRequestHandler actor termination caught by manager
[INFO] [...] [.../TestJesApiQueryManager-1262117937] PAPI request worker statusPoller2 has been removed and replaced by statusPoller3 in the pool of 1 workers

```

**Comment by cjllanwarne** [comment]
@mcovarr (finally!) ready for round 2

**Comment by mcovarr** [comment]
Guessing it's not a coincidence that all PAPI builds for this branch are broken?

**Comment by geoffjentry** [comment]
@mcovarr well the title of the pr **does** mention programmer error 😛

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/api/PipelinesApiRequestManager.scala [hunk 2]
```diff
@@ -177,46 +180,74 @@ class PipelinesApiRequestManager(val qps: Int Refined Positive, requestWorkers:
     }
   }
 
-  override protected def onFailure(terminee: ActorRef, throwable: => Throwable) = {
+  /*
+     Triggered by the StopAndLogSupervisor trait whenever a child (ie supevisee) terminates unexpectedly.
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/api/PipelinesApiRequestManager.scala]: Actually this comment needed a total refactor so I'm going to claim that the typo was actually clearly therefore a cleverly placed reminder.

**Pull request merged**